mautic/mautic
# IP2Location db4 db7 support

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N/A
| New feature? | **X**
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/192
| Related developer documentation PR URL | N/A
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/1164
| BC breaks? | N/A
| Deprecations? | N/A

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Added 2 new IP Lookup options. IP2Location local BIN lookup for DB4BIN and DB7BIN. These databases contain the ISP as well as the organization (domain) behind an IP address. I have also added these fields to the lead view so that one can see these values when they are present.

Note that IP2Location DDB4BIN and DB7BIN are not free databases. Also note that this is my first pull request here. If you have any feedback, please do let me know.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
N/A

#### Steps to test this PR:
1. Download the free sample of the DB7BIN [here](https://www.ip2location.com/databases/db7-ip-country-region-city-isp-domain)
2. Name the .BIN file "IP-COUNTRY-REGION-CITY-ISP-DOMAIN.BIN" and place it in app/cache/ip_data/
3. In the Backend->Configuration select "IP2Location Local Bin File (DB7BIN)" as your IP Lookup Service
4. Check if data gets added to incoming IP data and see if the fields are present in the Lead view

#### List deprecations along with the new alternative:
N/A

#### List backwards compatibility breaks:
N/A
